ravenframework/Optimizers/parentSelectors/parentSelectors.py

```diff
@@ -62,22 +68,25 @@ def rouletteWheel(population,**kwargs):
     roulettePointer = randomUtils.random(dim=1, samples=1)
     # initialize Probability
     counter = 0
-    if np.all(fitness.data>=0) or np.all(fitness.data<=0):
-      selectionProb = fitness.data/np.sum(fitness.data) # Share of the pie (rouletteWheel)
+    if np.all(fitness>=0) or np.all(fitness<=0):
+      selectionProb = fitness/np.sum(fitness) # Share of the pie (rouletteWheel)
     else:
       # shift the fitness to be all positive
-      shiftedFitness = fitness.data + abs(min(fitness.data))
+      shiftedFitness = fitness + abs(min(fitness))
       selectionProb = shiftedFitness/np.sum(shiftedFitness) # Share of the pie (rouletteWheel)
     sumProb = selectionProb[counter]
 
-    while sumProb < roulettePointer :
+    while sumProb <= roulettePointer :
       counter += 1
       sumProb += selectionProb[counter]
     selectedParent[i,:] = pop.values[counter,:]
     pop = np.delete(pop, counter, axis=0)
     fitness = np.delete(fitness,counter,axis=0)
   return selectedParent
 
+def countConstViolation(const):
+  return sum(1 for i in const if i < 0)
```

Comment:
dockstring missing.

Reply:
@JunyungKim, is this function even used? I think you no longer use it.
If so, please remove it